### ISSUE TITLE
target-size rule should ignore nested interactive controls.

### ISSUE THREAD

**Issue by WilcoFiers** [dequelabs/axe-core#3674]
Spotted this beauty on a video streaming site:

```html
  <a role="link" aria-label="play" tabindex="0">
    <button tabindex="-1" style="margin:1px;">Play</button>
  </a>
```

This is failing the rule because the link is considered to be largely obscured by the button inside of it. I think the solution here is that when we check for obscuring elements that we exclude descendants of the widget.

**Comment by padmavemulapati** [comment]
`target-size` rule off for now, as per the [ticket](https://github.com/dequelabs/axe-core/issues/3705).
As @straker suggessted, tried to fetch the results for `target-size` using `axe.run({ runOnly: 'target-size' });` but getting,
strange results: 
![Image](https://user-images.githubusercontent.com/47409579/195761876-7da62b48-8904-489e-abfb-fd0e919bd61c.png)

will wait for the dev confirmation.

**Comment by padmavemulapati** [comment]
@straker with the code snippet:

```
<a role="link" aria-label="play" tabindex="0">
    <button tabindex="-1" style="margin:1px;">Play</button>
  </a>
```
I am seeing `inapplicable` results , but it should pass `target-size`
![Image](https://user-images.githubusercontent.com/47409579/196844171-574ffbf6-af0e-426e-b74a-088b3d318bce.png)

**Comment by padmavemulapati** [comment]
Tried with the test snippet:
```
<main>
    <a role=“link” aria-label=“play” tabindex=“0” style=“padding: 5px;“>
      <button tabindex=“-1”>Play</button>
    </a>
  </main>
```
and now `target-size` is passing for the nested interactive controls



![Image](https://user-images.githubusercontent.com/47409579/197193254-86962eba-ac15-4259-8816-d5a1f81591f7.png)
